src/Stateless/Graph/UmlDotGraphStyle.cs

```diff
@@ -0,0 +1,196 @@
+    /// Generate DOT graphs in basic UML style
+    /// </summary>
+    public class UmlDotGraphStyle : IGraphStyle
+    {
+        /// <summary>Get the text that starts a new graph</summary>
+        /// <returns></returns>
+        override internal string GetPrefix()
+        {
+            return "digraph {\n"
+                      + "compound=true;\n"
+                      + "node [shape=Mrecord]\n"
+                      + "rankdir=\"LR\"\n";
+        }
+
+        internal override string FormatOneCluster(SuperState stateInfo)
+        {
+            string stateRepresentationString = "";
+            var sourceName = stateInfo.StateName;
+
+            string label = sourceName;
+
+            if ((stateInfo.EntryActions.Count > 0) || (stateInfo.ExitActions.Count > 0))
+            {
+                label += "\\n----------";
+                foreach (var entryAction in stateInfo.EntryActions)
+                    label += "\\nentry / " + entryAction;
+                foreach (var exitAction in stateInfo.ExitActions)
+                    label += "\\nexit / " + exitAction;
+            }
+
+            stateRepresentationString = "\n"
+                + $"subgraph {stateInfo.NodeName}" + "\n"
+                + "\t{" + "\n"
+                + $"\tlabel = \"{label}\"" + "\n";
+
+            foreach (State subState in stateInfo.SubStates)
+            {
+                stateRepresentationString += FormatOneState(subState);
+            }
+
+            stateRepresentationString += "}\n";
+
+            return stateRepresentationString;
+        }
+
+        /// <summary>
+        /// Generate the text for a single state
+        /// </summary>
+        /// <param name="state">The state to generate text for</param>
+        /// <returns></returns>
+        override internal string FormatOneState(State state)
+        {
+            if ((state.EntryActions.Count == 0) && (state.ExitActions.Count == 0))
+                return state.StateName + " [label=\"" + state.StateName + "\"];\n";
+
+            string f = state.StateName + " [label=\"" + state.StateName + "|";
+
+            List<string> es = new List<string>();
+            foreach (string entry in state.EntryActions)
+                es.Add("entry / " + entry);
+            foreach (string exit in state.ExitActions)
+                es.Add("exit / " + exit);
+
+            f += String.Join("\\n", es);
+
+            f += "\"];\n";
+
+            return f;
+        }
+
+        /// <summary>
+        /// Format all transitions for a graph
+        /// </summary>
+        /// <param name="transitions"></param>
+        /// <returns></returns>
+        override internal List<string> FormatAllTransitions(List<Stateless.Graph.Transition> transitions)
+        {
+            List<string> lines = new List<string>();
+
+            foreach (var transit in transitions)
+            {
+                string line = null;
+                StayTransition stay = transit as StayTransition;
```

Comment:
This... 

                StayTransition stay = transit as StayTransition;
                if (stay != null)

... can be written as

                if (transit is StayTransition stay)

Same goes for `fix` and `dyn`